parserator_web/static/js/index.js

```diff
@@ -1,2 +1,58 @@
 /* TODO: Flesh this out to connect the form to the API and render results
    in the #address-results div. */
+
+const form = document.querySelector('form');
+const address = form.elements['address'];
+const btnSubmit = document.querySelector('#submit');
+const addressResults = document.getElementById('address-results');
+const parseType = document.getElementById('parse-type');
+const tableBody = document.querySelector('tbody');
+
+const errorMessage = document.createElement('p');
+errorMessage.innerHTML = "Unable to parse this value due to repeated labels.";
+errorMessage.style.color = "red";
+errorMessage.style.fontWeight = "bold";
+errorMessage.style.display = "none";
+const parentDiv = form.parentNode;
+parentDiv.insertBefore(errorMessage, form);
+
+
+const getData = (query) => {
+   //make sure tableBody is cleared for each new request
+   tableBody.innerHTML = "";
+
+   fetch('http://localhost:8000/api/parse/?address=' + query)
```

Comment:
this works locally, but won't work on staging or production. one way to address this is to make the call a relative call like this

```javascript
fetch(api/parse/?address=' + query)
```

you might also like to uses [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) to handle building your parameter arguments.

```javascript
parameters = new URLSearchParams({address: query})
fetch(api/parse/? + parameters.toString())
```

Reply:
Ah, thank you, this makes a lot of sense! And I haven't had experience with URLSearchParams but after reading up on it, it looks like this is exactly what I'll be using from now on.